File: frontend/src/api/polish.ts
import apiClient from "@/api/client";

export type PolishMode = "expand" | "polish" | "abbreviate";

export interface PolishTask {
  id: string;
  mode: PolishMode;
  status: string;
  result_text: string | null;
  error_message: string | null;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
}

export async function createPolishTask(payload: {
  mode: PolishMode;
  text: string;
  requirements?: string | null;
  target_length?: number | null;
}) {
  const response = await apiClient.post<PolishTask>("/polish/tasks", payload);
  return response.data;
}

export async function getPolishTask(taskId: string) {
  const response = await apiClient.get<PolishTask>(
    `/polish/tasks/${encodeURIComponent(taskId)}`,
  );
  return response.data;
}

export async function cancelPolishTask(taskId: string) {
  const response = await apiClient.post<PolishTask>(
    `/polish/tasks/${encodeURIComponent(taskId)}/cancel`,
  );
  return response.data;
}

File: backend/schemas/polish.py
"""Pydantic schemas for VSTO-driven polish/expand/abbreviate tasks."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

PolishMode = Literal["expand", "polish", "abbreviate"]


class PolishTaskCreate(BaseModel):
    mode: PolishMode
    text: str = Field(min_length=1, max_length=20_000)
    requirements: str | None = Field(default=None, max_length=2_000)
    target_length: int | None = Field(default=None, ge=50, le=20_000)


class PolishTaskResponse(BaseModel):
    id: str
    mode: str
    status: str
    result_text: str | None
    error_message: str | None
    started_at: datetime | None
    completed_at: datetime | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
